src/Mapster.Tests/WhenRecordRegration.cs

```diff
@@ -0,0 +1,492 @@
+        public void AdaptClassIncludeStruct()
+        {
+            TypeAdapterConfig<SourceWithClass, DestinationWithStruct>
+                .ForType()
+                .Map(x => x.TestStruct, x => x.SourceWithStruct.TestStruct);
+
+            var source = new SourceWithClass
+            {
+                SourceWithStruct = new SourceWithStruct
+                {
+                    TestStruct = new TestStruct("A")
+                }
+            };
+
+            var destination = source.Adapt<DestinationWithStruct>();
+
+            destination.TestStruct.Property.ShouldBe("A");
+        }
+
+        /// <summary>
+        /// https://github.com/MapsterMapper/Mapster/issues/482
+        /// </summary>
+        [TestMethod]
+        public void AdaptClassToClassFromPrivateProperty_IsNotInstanse()
+        {
+            var _sourse = new TestClassPublicCtr(200);
+
+            var _destination = new TestClassProtectedCtrPrivateProperty(400, "Me");
+
+            var _result = _sourse.Adapt(_destination);
+
+
+
+
+            _destination.ShouldBeOfType<TestClassProtectedCtrPrivateProperty>();
+
+            _destination.X.ShouldBe(200);
+
+            _destination.Name.ShouldBe("Me");
+
+
+            object.ReferenceEquals(_destination, _result).ShouldBeTrue();
+        }
+
+
+        /// <summary>
+        /// https://github.com/MapsterMapper/Mapster/issues/427
+        /// </summary>
+        [TestMethod]
+        public void UpdateNullable()
+        {
+            var _sourse = new UserAccount("123", "123@gmail.com", new DateTime(2023, 9, 24));
+
+            var _update = new UpdateUser
+            {
+                Id = "123",
+
+            };
+
+            var configDate = new TypeAdapterConfig();
+            configDate.ForType<UpdateUser, UserAccount>()
+                .Map(dest => dest.Modified, src => new DateTime(2025, 9, 24))
+                .IgnoreNullValues(true);
+
+
+            _update.Adapt(_sourse, configDate);
+
+
+
+            var _sourseEmailUpdate = new UserAccount("123", "123@gmail.com", new DateTime(2023, 9, 24));
+
+            var _updateEmail = new UpdateUser
+            {
+                Email = "245@gmail.com",
+            };
+
+
+            var config = new TypeAdapterConfig();
+            config.ForType<UpdateUser, UserAccount>()
+
+                .IgnoreNullValues(true);
+
+
+            var _resultEmail = _updateEmail.Adapt(_sourseEmailUpdate, config);
+
+
+            _sourse.Id.ShouldBe("123");
+            _sourse.Created.ShouldBe(new DateTime(2023, 9, 24));
+            _sourse.Modified.ShouldBe(new DateTime(2025, 9, 24));
+            _sourse.Email.ShouldBe("123@gmail.com");
+
+
+            _sourseEmailUpdate.Id.ShouldBe("123");
+            _sourseEmailUpdate.Created.ShouldBe(new DateTime(2023, 9, 24));
+            _sourseEmailUpdate.Modified.ShouldBe(null);
+            _sourseEmailUpdate.Email.ShouldBe("245@gmail.com");
+
+
+
+        }
+
+
+
+
+        #region NowNotWorking
```

Comment:
Please remove unused/commented code. Feel free to uncomment the still failing tests, but add an ignore attribute to them.